sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -1059,7 +1071,12 @@ public SpannerWriteResult expand(PCollection<MutationGroup> input) {
                               spec.getBatchSizeBytes(),
                               spec.getMaxNumMutations(),
                               spec.getMaxNumRows(),
-                              spec.getGroupingFactor(),
+                              // Do not group on streaming unless explicitly set.
+                              spec.getGroupingFactor()
+                                  .orElse(
+                                      input.isBounded() == IsBounded.BOUNDED
+                                          ? DEFAULT_GROUPING_FACTOR
+                                          : 1),
```

Comment:
It would be nice if this were another constant. We could have `DEFAULT_GROUPING_FACTOR_BOUNDED` and `DEFAULT_GROUPING_FACTOR_UNBOUNDED`. It doesn't need to be done here, could be in a follow-up PR.

Reply:
Will do in a separate pr